src/gflownet/data/sampling_iterator.py

```diff
@@ -294,10 +294,12 @@ def __iter__(self):
                 for i in range(num_offline, len(trajs)):
                     if not is_valid[i].item():
                         continue
-                    try:
+                    if self.task.cand_type == "mols":
                         trajs[i]["smi"] = self.ctx.object_to_log_repr(trajs[i]["result"])
```

Comment:
If the string context implements `object_to_log_repr` the `if is mol` shouldn't be necessary.

Reply:
Thanks for noticing, not sure why I added that. Fixed!